### PR TITLE
feat(route): migrate japanpost to v2

### PR DESCRIPTION
**Pull request by tuzi3040**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/japanpost/track/225296644860/jp
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [x] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
  - [ ] EN / 英文文档
  - [ ] CN / 中文文档
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制 
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施? 
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
fix(route): japanpost: fix cheerio custom methods
fix(route): japanpost: fix manual timezone parse

### CODE REVIEW

File: assets/radar-rules.js [hunk 1]
```diff
@@ -171,23 +171,6 @@
         ],
         grs: [{ title: '研究生院通知', docs: 'https://docs.rsshub.app/routes/university#yun-nan-da-xue', source: '/*', target: '' }],
     },
-    'japanpost.jp': {
```
Comment (TonyRL) [inline assets/radar-rules.js:174]: Please remove the same entries in `lib/radar-rules.js` as well.

File: lib/v2/japanpost/maintainer.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    '/japanpost/track/:reqCode/:locale?': ['tuzi3040'],
```
Comment (TonyRL) [inline lib/v2/japanpost/maintainer.js:2]:
```suggestion
    '/track/:reqCode/:locale?': ['tuzi3040'],
```

File: lib/v2/japanpost/radar.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
+    'japanpost.jp': {
+        _name: '日本郵便',
+        'trackings.post': [
+            {
+                title: '郵便・荷物の追跡',
+                docs: 'https://docs.rsshub.app/routes/other#ri-ben-you-bian-you-bian-zhui-ji-サービス',
```
Comment (TonyRL) [inline lib/v2/japanpost/radar.js:7]:
```suggestion
                docs: 'https://docs.rsshub.app/routes/other#japanpost-ri-ben-you-bian',
```

File: lib/v2/japanpost/radar.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+                source: '/services/srv/search/direct',
+                target: (params, url) => {
+                    const reqCode = new URL(url).searchParams.get('reqCodeNo1').toUpperCase();
+                    const locale = new URL(url).searchParams.get('locale').toLowerCase();
+                    if ((reqCode.search(/^(?:\d{11,12}|[A-Z]{2}\d{9}[A-Z]{2})$/) === 0 && locale === 'ja') || locale === 'en') {
+                        return `/japanpost/track/${reqCode}/${locale}`;
+                    }
+                },
```
Comment (TonyRL) [inline lib/v2/japanpost/radar.js:15]: Please remove `rsshub=1` in the docs.

File: lib/v2/japanpost/templates/track_item_desc.art [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+{{ if officeItemList }}
+<br>
+{{ each officeItemList }}
+<br>{{ $value.officeType }} {{@ $value.officeName }} {{ $value.officeTel }}
+{{ /each }}
+{{ /if }}
+{{ if packageService }}
+<br>{{ serviceText }}{{ packageService }}
+{{ /if }}
+{{ /if }}
```
Comment (TonyRL) [inline lib/v2/japanpost/templates/track_item_desc.art:14]:
```suggestion
{{ if index === 0 }}
    {{ if officeItemList }}
        <br>
        {{ each officeItemList }}
            <br>{{ $value.officeType }} {{@ $value.officeName }} {{ $value.officeTel }}
        {{ /each }}
    {{ /if }}
    {{ if packageService }}
        <br>{{ serviceText }}{{ packageService }}
    {{ /if }}
{{ /if }}
```

File: lib/v2/japanpost/radar.js (hunk 4 above)
Comment (tuzi3040) [reply lib/v2/japanpost/radar.js:15]: You might refer to `rssbud="1"` in the docs?
I've tested on my device, and RSSBud functions normally with this radar rule.

**Pull request merged**